integration/test/basic/basic_test.go

```diff
@@ -63,25 +59,57 @@ func TestChartConfigChartsInstalled(t *testing.T) {
 		return
 	}
 
-	guestNamespace := "giantswarm"
 	guestG8sClient := g.G8sClient()
 
 	// Wait for chart configs as they may not have been created yet.
 	err := waitForChartConfigs(guestG8sClient)
 	if err != nil {
 		t.Fatalf("could not get chartconfigs %v", err)
 	}
+}
+
+func TestChartConfigPatchDeploy(t *testing.T) {
+	// These versions have no chartconfigs so we return early.
+	clusterOperatorVersion := os.Getenv("CLOP_VERSION_BUNDLE_VERSION")
+	if clusterOperatorVersion == "0.1.0" || clusterOperatorVersion == "0.2.0" {
+		return
+	}
+
+	guestG8sClient := g.G8sClient()
 
 	chartConfigList, err := guestG8sClient.CoreV1alpha1().ChartConfigs(guestNamespace).List(metav1.ListOptions{})
 	if err != nil {
 		t.Fatalf("could not get chartconfigs %v", err)
 	}
-
-	// At least 1 chartconfig is expected. Actual number depends on how many
-	// components have been migrated.
+	// At least 1 chartconfig is required.
 	if len(chartConfigList.Items) == 0 {
```

Comment:
Shouldn't the check be more strict here, i.e. checking if this isn't `1`? Later in the test `chartConfigName` is used, and if this number is bigger than `1` then the test is indeterministic.

Reply:
I think checking there are chartconfigs is good. I don't think we should check the count because it changes often. Good point that the test should be deterministic. Lets check for a specific chartconfig that is always there.